### DISCUSSION TITLE
How do I download items inside a folder shared with me?

### DISCUSSION THREAD

**Discussion by estevanmendes** [O365/python-o365#557] [Q&A]
I am trying to find a folder by its path, but I am not finding. I think it's because I am not the owner of the files, it was only shared with me. Although, when I search for the name of the folder a can find it and the files inside it. 

To workaround this I was downloading every item that is not a folder and has the extension that I am searching for through this code below.
However, I am only being able to download files that belongs to my personal OneDrive library.
```
from O365 import Account, MSGraphProtocol
import os



id=myId
client_secret=myClientSecret

credentials = (id,client_secret)


protocol = MSGraphProtocol(api_version='beta')  # MSGraphProtocol defaults to v1.0 api version
account = Account(credentials, protocol=protocol)

account = Account(credentials=credentials)

storage = account.storage()  # here we get the storage instance that handles all the storage options.

# list all the drives:
drives = storage.get_drives()

# get the default drive
my_drive = storage.get_default_drive()  # or get_drive('drive-id')

shared_folders=my_drive.get_shared_with_me()

files=my_drive.search('Efetivo Carcerário -')

for item in files:
    if 'File'==str(item).split(':')[0]:
        if 'xlsx'==str(item).split('.')[-1]:
           print(item.download())
           
```

**Comment by alejcas** [comment]
You should use:

`drive.get_shared_with_me`

You can filter some items although notice that Ms graph limits the filters you can apply.

Then test each item returned to see if it's a folder and the name is the same as expected. Then you can get the folder and download it's items.

**Comment by cgarmstr** [comment]
Did you get this sorted out? I've been trying to do the same thing.

When I print out all the attributes of a shared folder called 'LVO', is_folder=False and is_file=True. However if I get remote=item.remote_item, is_folder=True and is_file=False. Unfortunately, when I try and do remote.get_items() it fails with: Item does not exist. Oddly, remote.child_count returns the right answer.

Any clues appreciates.

from O365 import Account
client_id = '...'
client_secret_id = '...'
client_secret_value = '...'
credentials = (client_id, client_secret_value)
account = Account(credentials)
storage = account.storage()
my_drive = storage.get_default_drive()
shared=my_drive.get_shared_with_me()
for item in shared:
    if item.name == 'LVO':  # Just get one shared folder called 'LVO'
        print(f' {item.name:15}')  # print the mime type
        attrs = dir(item)
        print(attrs)
        for a in attrs:
            print(f'{a:30} {item.__getattribute__(a)}')

        remote = item.remote_item
        remote_attrs = dir(remote)
        print('\n Remote item attrs')
        for a in remote_attrs:
            print(f'{a:30} {remote.__getattribute__(a)}')
        print('\n Try getting remote attributes')
        for a in remote_attrs:
            result = item.__getattribute__(a)
            print(f'{a:30} {item.__getattribute__(a)}')
       print(f'child_count = {remote.child_count}') # 13
      remote_folder_items = remote.get_items()   # Fails with Item does not exist
